flow360/component/simulation/models/validation/validation_bet_disk.py

```diff
@@ -0,0 +1,133 @@
+"""
+validation BETDisk
+"""
+
+
+# pylint: disable=missing-function-docstring
+def _get_bet_disk_name(bet_disk):
+    disk_name = "one BET disk" if bet_disk.name is None else f"the BET disk {bet_disk.name}"
+    return disk_name
+
+
+def _check_bet_disk_initial_blade_direction(bet_disk):
+    disk_name = _get_bet_disk_name(bet_disk)
+    if bet_disk.blade_line_chord > 0 and bet_disk.initial_blade_direction is None:
+        raise ValueError(
+            f"On {disk_name}, the initial_blade_direction"
+            " is required to specify since its blade_line_chord is non-zero."
+        )
+    if bet_disk.initial_blade_direction is not None and bet_disk.blade_line_chord == 0:
+        raise ValueError(
+            f"On {disk_name}, the blade_line_chord has to be positive"
+            " since its initial_blade_direction is specified."
+        )
+    return bet_disk
+
+
+def _check_bet_disk_alphas_in_order(bet_disk):
+    disk_name = _get_bet_disk_name(bet_disk)
+    alphas = bet_disk.alphas
+    if alphas != sorted(alphas):
+        raise ValueError(f"On {disk_name}, the alphas are not in increasing order.")
+    return bet_disk
+
+
+def _check_has_duplicate_in_one_radial_list(radial_list):
+    existing_radius = set()
```

Comment:
This is tricky if BETDisk field become dimensional. We will revisit once that happens.